src/cpu/x64/brgemm/jit_brgemm_amx_uker.cpp

```diff
@@ -312,18 +312,13 @@ struct jit_brgemm_amx_uker_base_t : public jit_base_brgemm_kernel_t {
         bd_iteration_t *similar {nullptr};
         Label lstart;
 
-        bool operator==(const dim_iteration_t &_rhs) const override {
-            // `downcast` will catch a type mismatch in debug mode.
-            // Note: it supports only a pointer type so far.
-            const bd_iteration_t &rhs
-                    = *utils::downcast<const bd_iteration_t *>(&_rhs);
-            bool res = dim_iteration_t::operator==(rhs)
-                    && A_shift == rhs.A_shift && C_shift == rhs.C_shift
-                    && D_shift == rhs.D_shift && bd_mask == rhs.bd_mask
+        bool operator==(const bd_iteration_t &rhs) const {
```

Comment:
Since this class is not polymorphic can you please make `dim_iteration_t`'s destructor non virtual?